app/lib/clients/vault/certificate.rb

```diff
@@ -38,36 +45,59 @@ def cert_engine_type
       end
 
       def enable_ca
-        # if mount exists, assume configuration is done
-        if client.sys.mounts.key?(intermediate_ca_mount.to_sym)
-          return false
-        end
-
-        # create the mount
+        # create the intermediate mount
         enable_engine(intermediate_ca_mount, cert_engine_type)
-        true
+      end
+
+      def configure_root_ca
+        return if client.sys.mounts.key?(root_ca_mount.to_sym)
+
+        # enable engine
+        enable_engine(root_ca_mount, cert_engine_type)
+
+        # generate root certificate
+        root_cert = client.logical.write("#{root_ca_mount}/root/generate/internal",
+                                        common_name: "astral.internal",
+                                        issuer_name: root_ca_ref,
+                                        ttl: "87600h").data[:certificate]
+        # save the root certificate
+        File.write("tmp/#{root_ca_ref}.crt", root_cert)
```

Comment:
This doesn't seem to be used any where.  I assume you are just saving it to a known location where you can find it manually if needed, right?

Reply:
yes -- i am not sure it's really needed since it can be retrieved as needed from vault -- and this does not seem to include the private key

Reply:
okay there's no private key because these are generated as `internal` where Vault keeps the private key and we never need to know it. Generating the cert with `exported` path will respond with private key as well as cert.